### ISSUE TITLE
[eng quality] implement git hook for rustfmt

### ISSUE THREAD

**Issue by pavitthrap** [stacks-network/stacks-blockchain#3570]
Create a pre-commit hook to check the formatting of the Rust files. 
A description of how to do so can be found here: (https://eugene-babichenko.github.io/blog/2018/11/08/rustfmt-git-hook/)

Since the `.git/hooks` is not in version control, the hook should be stored in a different folder, like `.githooks`, so that all developers have the same hook. (https://www.atlassian.com/git/tutorials/git-hooks) 

This issue is especially important because of the new check for import order in the CI, implemented in #3556

**Comment by jcnelson** [comment]
I'd prefer that these git hooks remain locally configured.  I'm generally not a fan of code that tries to script other peoples' `git` behavior.  It's a great way for inserting back doors, for example.

**Comment by igorsyl** [comment]
Agreed, we just need instructions on enabling the hooks for new devs.

**Comment by kantai** [comment]
I agree that I'd prefer to leave this as something developers can locally configure. It might be helpful to include a sample in the `CONTRIBUTING.md` like:

```md

### Recommended developer setup

#### Recommended githooks


#!/bin/sh
# Redirect output to stderr.
exec 1>&2

cargo fmt -q --all -- --check

CARGO_FMT_RESULT=$?

if [ $CARGO_FMT_RESULT -eq 1 ]
then
    echo 'rustfmt failed: run "cargo fmt --all"'
fi

exit $CARGO_FMT_RESULT
```

**Comment by pavitthrap** [comment]
closed by #3586
